compiler/rustc_session/src/parse.rs

```diff
@@ -56,9 +56,9 @@ impl GatedSpans {
     }
 
     /// Prepend the given set of `spans` onto the set in `self`.
-    pub fn merge(&self, mut spans: FxHashMap<Symbol, Vec<Span>>) {
+    pub fn merge(&self, mut spans: FxIndexMap<Symbol, Vec<Span>>) {
         let mut inner = self.spans.borrow_mut();
-        for (gate, mut gate_spans) in inner.drain() {
+        for (gate, mut gate_spans) in inner.drain(..) {
```

Comment:
As this just puts the contents of the drained hashmap into another one, I don't see a way that the iteration order can affect the resulting map. So we could also add a local `allow` here but replacing the map seemed preferable. Though, I'm not sure if this has any performance implications.